2-multitenant/modules/env_baseline/main.tf

```diff
@@ -158,3 +174,42 @@ module "gke" {
 
   deletion_protection = false # set to true to prevent the module from deleting the cluster on destroy
 }
+
+module "gke-autopilot" {
+  source  = "terraform-google-modules/kubernetes-engine/google//modules/beta-autopilot-private-cluster"
+  version = "~> 31.0"
+
+  for_each = var.cluster_type == "AUTOPILOT" ? data.google_compute_subnetwork.default : {}
+  name     = "cluster-${each.value.region}-${var.env}"
+
+  project_id          = local.cluster_project_id
+  regional            = true
+  region              = each.value.region
+  network_project_id  = regex(local.projects_re, each.value.id)[0]
+  network             = regex(local.networks_re, each.value.network)[0]
+  subnetwork          = each.value.name
+  ip_range_pods       = each.value.secondary_ip_range[0].range_name
+  ip_range_services   = each.value.secondary_ip_range[1].range_name
+  release_channel     = var.cluster_release_channel
+  gateway_api_channel = "CHANNEL_STANDARD"
+
+  security_posture_vulnerability_mode = "VULNERABILITY_ENTERPRISE"
+  enable_cost_allocation              = true
+
+  fleet_project = local.cluster_project_id
+
+  identity_namespace = "${local.cluster_project_id}.svc.id.goog"
+
+  # TODO: https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/issues/2029
+  #enable_binary_authorization = true
```

Comment:
the github issue seems to be closed, should this be uncommented?

Reply:
:) Yeah, the creation of this PR pre-date the close of that issue.

Reply:
Updated to the version with binauthz support.